docs/conf.py

```diff
@@ -126,11 +137,61 @@
 # If true, keep warnings as "system message" paragraphs in the built documents.
 # keep_warnings = False
 
+##########################################################################
+## Extension Configuration
+##########################################################################
+
 # If true, `todo` and `todoList` produce output, else they produce nothing.
 todo_include_todos = False
 
+# Auto-plot settings either as extension or (file format, dpi)
+plot_formats = [
+    'png',
+    'pdf',
+    # ('hires.png', 350),
+]
+
+# By default, include the source code generating plots in documentation
+plot_include_source = True
+
+# Whether to show a link to the source in HTML.
+plot_html_show_source_link = True
+
+# Code that should be executed before each plot.
+plot_pre_code = (
+    "import numpy as np\n"
+    "import matplotlib.pyplot as plt\n"
+    "from yellowbrick.datasets import *\n"
+)
```

Comment:
Note that this directive ensures that `plt`, `np`, and all of the dataset loaders are available during the `.. plot::` directive, so they don't have to be specifically imported. This might also help with the code splits thing, which were generally about loading datasets.